radsat-sk/operation/subsystems/RPdb.c

```diff
@@ -0,0 +1,330 @@
+		0x10E224,		// Output Current of Battery Bus
+		0x10E214,		// Output Current of 5V Bus
+		0x10E204		// Output Current of 3.3V Bus
+};
+
+/**
+ * Command for PDB Temperature Reading
+ */
+static uint32_t pdbTemperatureCommandBytes = 0x10E308;
+
+/**
+ * Command for Resetting PDB Watchdog (0x2200)
+ */
+static uint8_t pdbWatchdogResetCommand[2] = {0x22, 0x00};
+
+
+/***************************************************************************************************
+                                       PRIVATE FUNCTION STUBS
+***************************************************************************************************/
+
+static int pdbTalk(uint8_t* command, uint8_t* response);
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+/**
+ * Get the Irradiance (W/m^2) value from the sun sensor on each face of the CubeSat
+ *
+ * @param A pointer to a sun_sensor_status_t object, which will be filled with the converted ADC data for each sun sensor
+ * @return an error code, or 0 if no error
+ */
+int getSunSensorData(sun_sensor_status_t* sunData) {
+
+	// Create temporary variables for sending and receiving I2C data
+	uint8_t command[PDB_TELEM_COMMAND_LENGTH] = {0};
+	uint8_t response[PDB_RESPONSE_LENGTH] = {0};
+
+	// Create a temporary array for calculated sun data before transferring into the structure sunData
+	float convertedData[NUM_SUN_SENSORS] = {0};
+
+	// Send 6 commands to get each of the sun sensor's data
+	for (int i = 0; i < NUM_SUN_SENSORS; i = i + 1) {
+
+		// Load command into output buffer
+		memset(command, 0, sizeof(command));
+		memcpy(command, &pdbSunSensorCommandBytes[i], PDB_TELEM_COMMAND_LENGTH);
+
+		memset(response, 0, sizeof(response));
+
+		int error = pdbTalk(command, response);
+
+		if (error != SUCCESS) {
+			return error;
+		}
+
+		memset(&convertedData[i], 0, sizeof(convertedData[i]));
+		memcpy(&convertedData[i], response, PDB_RESPONSE_LENGTH);
+
+		convertedData[i] = ADC_COUNT_TO_IRRADIANCE * ((float)convertedData[i]);
+	}
+
+	// Now store all of the calculated data into the proper slot in the sunData structure
+	sunData->xPos = convertedData[3]; 	// SA2B
+	sunData->xNeg = convertedData[2]; 	// SA2A
+	sunData->yPos = convertedData[0]; 	// SA1A
+	sunData->yNeg = convertedData[1]; 	// SA1B
+	sunData->zPos = convertedData[5]; 	// SA3B
+	sunData->zNeg = convertedData[4]; 	// SA3A
+	// Could change these numbers into an ENUM with the names of the connections
+
+	return SUCCESS;
+}
+
+
+/**
+ * Request and store all of the relevant data (voltage, current, temperature, sun sensor data)
+ *  	from the PDB board
+ *
+ * @param A pointer to an PDBStatus object containing all of the telemetry for the PDB system
+ * @return either 0 or an error code
+ */
+int getPdbTelemetry(pdb_status_t* dataStorage) {
```

Comment:
Same comment as above; `pdbTelemetry` would be better in my opinion (the "get" is implied)